### DISCUSSION TITLE
Sonarr LXC script : Unable to add root folder Folder is not writable by user sonarr

### DISCUSSION THREAD

**Discussion by ghost** [tteck/Proxmox#1316] [Help Needed] [answered]
Hi all,

I Created a Privileged container with the LXC script and when trying to map a mounted drive, I get the following error:
Unable to add root folder Folder is not writable by user sonarr

I just installed Radarr via the exact same way without an issue, same for SABnzb
Mapped drives are visible and accessible 

![image](https://user-images.githubusercontent.com/89245209/230794444-093bf648-d30f-43cc-834e-65245e2b5ec5.png)

![image](https://user-images.githubusercontent.com/89245209/230794462-bb30cc31-fe3a-42f1-be22-aaae4dac9c6f.png)

![image](https://user-images.githubusercontent.com/89245209/230794520-22a9490e-d2c9-4b29-86b7-473193fb7724.png)

![image](https://user-images.githubusercontent.com/89245209/230794585-f545d8f4-fc49-4aaa-a77d-e0b7652bc3e7.png)

Also noticed the Sonarr CT is using the dev branch - might not be related. Radarr one is production version

![image](https://user-images.githubusercontent.com/89245209/230794697-295a6e1e-6f95-4764-b1e5-f37a8b55bd18.png)
![image](https://user-images.githubusercontent.com/89245209/230794719-3736f0d7-4125-4cfe-a951-d6ccf750defd.png)

Anyone know how to go from here or experiencing the same issues? Kind of new with Linux and Proxmox but eager to learn.

Thanks in advance

**Comment by AAH134** [comment]
I fixed it I remember this sonar have an issue, I will trace back what I did, and drop it in here. This took me a whole day to fix 2 months ago

**Reply by AAH134** [reply]
Ok here is what worked as I remember,

Everything I did, never worked, until I did unstalling it from the sonarr console, and reinstall


sudo apt remove sonarr
Then
sudo apt install sonarr

**Reply by ghost** [reply]
> sudo apt install sonarr

Appreciate your answer and willingness to help, unfortunatly this has not resolved the issue.
After running the install command - I get prompted to select a user and usergroup. Should I use another group or user instead of the standard sonarr one?

**Reply by tteck** [reply]
You could try `root:root`  🤷🏻‍♂️

**Reply by ghost** [reply]
> You could try `root:root` 🤷🏻‍♂️

Appreciate your reply, still a bit new to this so to confirm - I should at this as an argument to the command line @AAH134 suggested?
Thanks for your support!!

**Reply by ghost** [reply, marked as answer]
`dpkg-reconfigure sonarr` did the job - remapped to root

**Reply by Drallas** [reply]
Thanks, worked, but only on a fresh install for me.

**Reply by tteck** [reply]
Use `dpkg-reconfigure -plow sonarr` and set the following; 
Sonarr user: **root**, Sonarr group: **root** and Sonarr umask: **0000**

So, a umask of 0000 sets the default permissions for new files and directories to be fully permissive, with read and write permissions

**Reply by Drallas** [reply]
Interesting, not sure if I want Arr apps to do that!?

I set the download (Plex host) folder fully permissive (777), anything can modify there, it’s only a temp cache imho..

Since my Arr LXCs are unprivileged, the host folders my bind mounts point to, need to have 10000:10000, with an unmask:0022. 

Deluge and SABnzbd restrict file access too much, so I let Arr Media Management also apply 0022 to all imports. Effectively only the LXC root and host root users can modify imported files.

Plex seems to need to be run privileged for transcoding to work, that added to the challenge. Plex Media needs at least Read Access for UG(Others) it’s bind Mount is anyway set to ,ro!

I’m not sure what the best practice is, but this works for me. 

Btw users that need access, I plan to give permission via ACL’s.

Curious how others do all this!?

**Reply by Rifleman75** [reply]
I'm having the exact same issue but when I try to run dpkg-reconfigure sonarr it just says that sonarr isn't installed. 
 
I've been fighting this for two days now.

**Reply by tteck** [reply]
Sonarr is not installed via a `.deb` package any longer, that's why `dpkg-reconfigure sonarr` doesn't work.

**Reply by Rifleman75** [reply]
Is there a new way to fix this issue then?

**Reply by tteck** [reply]
Make sure you have a backup.

You can execute the installer (within the LXC console) from Sonarr to see if that gets you straightened out. 🤷🏻‍♂️ 
```
bash -c "$(wget -qLO - https://github.com/Sonarr/Sonarr/raw/develop/distribution/debian/install.sh)"
```

**Reply by Rifleman75** [reply]
That results in the console crashing... same with running update.

**Reply by tteck** [reply]
You have read https://github.com/tteck/Proxmox/discussions/1738#discussioncomment-8005107?

**Reply by Rifleman75** [reply]
This finally ran after a LONG time which is why I thought it was crashing. It was intermittent in the options it would give me when it would run and I could never get it to work correctly.

**Comment by JanRi3D** [comment]
I think you need top give the permisson to every User to make it easier with "chmod 777 /mnt/Media or Something Like that than it should work

**Reply by ghost** [reply]
In the LXC container / console you mean? Just tried this but without any luck :(

**Reply by AAH134** [reply]
I have tried that before but trust me there is a bug or issue with the deployment script. I had to reinstall to see it working

**Reply by tteck** [reply]
> trust me there is a bug or issue with the deployment script

@AAH134 Did you review the script in order to assert that? Can you provide further details?

**Reply by AAH134** [reply]
I dont mean it is a bug in this script but could be in apt install sonarr it self, or the version it is using.
have you tried the buster instead of buster-develop as following?
echo "deb https://apt.sonarr.tv/debian **buster** main" | sudo tee /etc/apt/sources.list.d/sonarr.list

**Comment by escaflown23** [comment]
Just wanted to say that I had the exact same issue and running `dpkg-reconfigure sonarr` fixed it for me as well.
